### PR TITLE
Make promotions store aware

### PR DESCRIPTION
**Pull request by RyanMacG**

### CODE REVIEW

**Comment by RyanMacG** [comment]
see #2 for context

File: db/migrate/20161116122026_store_id_for_promotions.rb [hunk 1]
```diff
@@ -0,0 +1,19 @@
+class StoreIdForPromotions < ActiveRecord::Migration
+  def self.up
+    if table_exists?('promotions')
```
Comment (SebAshton) [inline db/migrate/20161116122026_store_id_for_promotions.rb:3]: Is this needed? Tables should always be `spree_` prefixed

File: spec/models/spree/promotion_spec.rb [hunk 2]
```diff
@@ -0,0 +1,18 @@
+require 'spec_helper'
+
+describe Spree::Promotion do
+  before(:each) do
+    @store = FactoryGirl.create(:store)
```
Comment (SebAshton) [inline spec/models/spree/promotion_spec.rb]: Can this be a `let`, same for `@promotions` below

**Review by SebAshton — CHANGES_REQUESTED** [review]

File: spec/models/spree/promotion_spec.rb (hunk 2 above)
Comment (RyanMacG) [reply spec/models/spree/promotion_spec.rb]: Sure, was following the convention of the gem

File: db/migrate/20161116122026_store_id_for_promotions.rb (hunk 1 above)
Comment (RyanMacG) [reply db/migrate/20161116122026_store_id_for_promotions.rb:3]: All the other migrations had it so I wasn't sure

**Review by SebAshton — APPROVED** [review]

**Pull request merged**